backend/api_calls.py

```diff
@@ -728,6 +728,64 @@ def AddPlayerToChat(request, firebase):
   results.append(firebase.put('/games/%s/chatRooms' % game, chat, True))
 
 
+def Infect(request, firebase):
+  """Infect a player(otherPlayerId) and update access to chat rooms accordingly.
+
+  Validation:
+    playerId:
+    otherPlayerId:
+    gameId:
+
+    
+  """
+  results = []
+  valid_args = ['infectionId', 'playerId', 'otherPlayerId', 'gameId']
+  required_args = ['otherPlayerId', 'gameId']
+  ValidateInputs(request, firebase, required_args, valid_args)
+  playerId = request['playerId']
+  gameId = request['gameId']
+
+  if playerId and firebase.get('/games/%s/players/%s' % (gameId, playerId), None)['canInfect'] != "yes":
```

Comment:
1. No need to test for playerId. Validate() did that.
2. You can change None to 'canInfect' and ask Firebase for only that value.
3.￼ canInfect is a bool, not a string.